hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -410,6 +414,12 @@ private void handleBootstrapEvent(WriteMetadataEvent event) {
     if (Arrays.stream(eventBuffer).allMatch(evt -> evt != null && evt.isBootstrap())) {
       // start to initialize the instant.
       initInstant(event.getInstantTime());
+    } else if (event.isBootstrap() && subtaskResetStatus[event.getTaskID()] == 1
+        && !Arrays.stream(subtaskResetStatus).allMatch(st -> st == 1)) {
+      // handle partial failover, if only a subset of subtasks send bootstrap event, hence not all subtasks need to restore,
+      // then these subtasks can reuse previous instant to write
```

Comment:
The full re-start failover is included in this check condition :)

Reply:
> The full re-start failover is included in this check condition :)

Do you mean included in the first condition if (Arrays.stream(eventBuffer).allMatch(evt -> evt != null && evt.isBootstrap())) ?